src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -4922,17 +4915,24 @@ void handleMemberInitializer(IMemberInitializerOperation memberInitializer)
                         _evalStack.Push(Visit(arrayReference.ArrayReference));
                         return (success: true, indicies);
 
-                    case OperationKind.DynamicIndexerAccess: // PROTOTYPE(dataflow): For now handle as invalid case to enable other test scenarios
-                    case OperationKind.None: // PROTOTYPE(dataflow): For now handle as invalid case to enable other test scenarios.
-                                             //                      see Microsoft.CodeAnalysis.CSharp.UnitTests.IOperationTests.ObjectCreationWithDynamicMemberInitializer_01
-                    case OperationKind.Invalid:
-                        return (success: false, arguments: ImmutableArray<IOperation>.Empty);
+                    case OperationKind.DynamicIndexerAccess:
+                        var dynamicIndexer = (IDynamicIndexerAccessOperation)instance;
+                        ImmutableArray<IOperation> arguments = dynamicIndexer.Arguments.SelectAsArray(argument =>
+                        {
+                            int captureId = VisitAndCapture(argument);
+                            return (IOperation)GetCaptureReference(captureId, argument);
+                        });
+                        _evalStack.Push(Visit(dynamicIndexer.Operation));
+                        return (success: true, arguments);
+
 
                     default:
-                        // PROTOTYPE(dataflow): Probably it will be more robust to handle all remaining cases as an invalid case.
-                        //                      Likely not worth it throwing for some edge error case. Asserting would be good though
-                        //                      so that tests could catch the things we though are impossible.
-                        throw ExceptionUtilities.UnexpectedValue(instance.Kind);
+                        // As in the assert in handleInitializer, this assert documents the operation kinds that we know go through this path,
+                        // and it is possible others go through here as well. If they are encountered, we simply need to ensure
+                        // that they don't have any interesting semantics in object or collection initialization contexts and add them to the
+                        // assert.
+                        Debug.Assert(instance.Kind == OperationKind.Invalid || instance.Kind == OperationKind.None);
+                        return (success: false, arguments: ImmutableArray<IOperation>.Empty);
```

Comment:
I don't follow why `success` is false here. It seems like you should either return true or change the assert to a `throw`.

Reply:
This method attempts to push the instance onto the stack with special handling. If we're in this case, no special handling is required. We're not 100% confident that we've tested all error scenarios that will run through this code path, but we don't want to fail to build the graph in that scenario. That's why this method is named `tryPushTarget`, because there are scenarios where we cannot.

---
In reply to: [193503389](https://github.com/dotnet/roslyn/pull/27519#discussion_r193503389) [](ancestors = 193503389)